### PR TITLE
Lesson 38

### PR DESCRIPTION
**Pull request by ArtyomLobanok**

### CODE REVIEW

File: src/components/FormSingUp/Form.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import {useState} from 'react';
+
+const Form = ({title, handleClick}) => {
+    const [email, setEmail] = useState('');
+    const [pass, setPass] = useState('');
+
+    return (
+        <div>
```
Comment (romos21) [inline src/components/FormSingUp/Form.jsx]: why div tag instead of form?

File: src/components/FormSingUp/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            .then(({user}) => {
+                console.log(user);
+                dispatch(setUser({
+                    email: user.email,
+                    id: user.uid,
+                    token: user.accessToken,
+                }));
+                navigate('/');
+            })
+            .catch(() => alert('Invalid user!'))
```
Comment (romos21) [inline src/components/FormSingUp/Login.jsx]: use any popup instead of alert

File: src/components/FormSingUp/Form.jsx [hunk 3]
```diff
@@ -1,22 +1,26 @@
...
+        <div className="signIn">
+            <label>
+                Email address
+            </label>
             <input
                 type="email"
                 value={email}
                 onChange={(e) => setEmail(e.target.value)}
-                placeholder="email"
             />
```
Comment (romos21) [inline src/components/FormSingUp/Form.jsx]: connect label with input

Comment (romos21) [reply src/components/FormSingUp/Form.jsx]: using for attribute or wrapping input into label

File: src/components/FormSingUp/Form.jsx (hunk 1 above)
Comment (ArtyomLobanok) [reply src/components/FormSingUp/Form.jsx]: did this

File: src/components/FormSingUp/Login.jsx (hunk 2 above)
Comment (ArtyomLobanok) [reply src/components/FormSingUp/Login.jsx]: did this

File: src/components/FormSingUp/Form.jsx (hunk 3 above)
Comment (ArtyomLobanok) [reply src/components/FormSingUp/Form.jsx]: did this

File: src/components/AvailableHotelsSlider/AvailableHotelsSlider.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
+import {useSelector} from "react-redux";
+import Slider from "react-slick";
+import HotelCard from "../HotelCard/HotelCard";
+import {Flex, SliderArrow, SliderButtonNext, SliderButtonPrev, SliderWrapper, Text} from "../Styled-components/Global";
+import React from "react";
+
+const AvailableHotelsSlider = () => {
+    const hotelCards = useSelector(state => state.hotelsLoadReducer.hotelData)
+    console.log(hotelCards)
+    const SampleNextArrow = (props) => {
```
Comment (ghost) [inline src/components/AvailableHotelsSlider/AvailableHotelsSlider.js]: you can write `const SampleNextArrow = ({className, onClick}) => `

File: src/components/AvailableHotelsSlider/AvailableHotelsSlider.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
+import {useSelector} from "react-redux";
+import Slider from "react-slick";
+import HotelCard from "../HotelCard/HotelCard";
+import {Flex, SliderArrow, SliderButtonNext, SliderButtonPrev, SliderWrapper, Text} from "../Styled-components/Global";
+import React from "react";
+
+const AvailableHotelsSlider = () => {
+    const hotelCards = useSelector(state => state.hotelsLoadReducer.hotelData)
+    console.log(hotelCards)
```
Comment (ghost) [inline src/components/AvailableHotelsSlider/AvailableHotelsSlider.js]: why do we need it?

File: src/components/AvailableHotelsSlider/AvailableHotelsSlider.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+                    <SliderArrow margin='0 0 0 10px' width='9px' height='21px'>
+                        <use href="#Arrow"></use>
+                    </SliderArrow>
+                </SliderButtonNext>
+            </div>
+        );
+    }
+
+    const SamplePrevArrow = (props) => {
+        const {className, onClick} = props;
```
Comment (ghost) [inline src/components/AvailableHotelsSlider/AvailableHotelsSlider.js]: the same

File: src/components/FormOfSeach/Form.js [hunk 7]
```diff
@@ -1,25 +1,45 @@
...
+        e.preventDefault()
+        dispatch(hotelsLoad(params));
+    }
 
-function Form ({setSearchData}) {
+    const hotelArray = useSelector(state => state.hotelsLoadReducer.hotelData)
+    useEffect(
+        () => {
+            if (hotelArray.length !== 0) {
+                window.scrollTo({
```
Comment (ghost) [inline src/components/FormOfSeach/Form.js:27]: optional: lets move it into custom hook

File: src/components/FormOfSeach/Form.js [hunk 8]
```diff
@@ -1,25 +1,45 @@
...
-                    <button type='submit' className="form__button search">Search</button>
-                </form>
-            </div>
+        <FormHeaderWrapper>
+            <StyledForm id="mainForm">
+                <FirstForm/>
+                <MyDataPicker/>
+                <FormFieldThird/>
+                <SearchBtn
+                    onClick={handleChange} type='submit'>Search</SearchBtn>
```
Comment (ghost) [inline src/components/FormOfSeach/Form.js]: type='submit' - used by default, no needs to pass it

File: src/components/FormOfSeach/FormFieldSearchHotel/FormFieldSearchHotel.js [hunk 9]
```diff
@@ -1,46 +1,67 @@
...
+    const dispatch = useDispatch();
+    const handleChange = (value) => {
+        dispatch(inputText(value))
     }
+    const searchText = useSelector(state => state.formReducer.search);
+    useEffect(
+        () => {
+            if (searchText.length !== 0 && (window.scrollY > 2400)) {
+                setIsActive(true)
+                window.scrollTo({
```
Comment (ghost) [inline src/components/FormOfSeach/FormFieldSearchHotel/FormFieldSearchHotel.js:37]: this is the reason why I asked to move into hook - you use it in multiple places

File: src/components/FormOfSeach/FormFieldSearchHotel/FormFieldSearchHotel.js [hunk 10]
```diff
@@ -1,46 +1,67 @@
...
-    useEffect(() => {
-        setSearchData(values)
-    }, [values])
     return (
-        <div className="first__input">
-            <input
-                name='nameOfSearch'
+        <StyledInputWrapper position='relative' maxWidth='467px'>
+            <StyledLabelFist active={isActive}>
+                Your destination or hotel
```
Comment (ghost) [inline src/components/FormOfSeach/FormFieldSearchHotel/FormFieldSearchHotel.js]: I'd recommend to store all your static strings in separate files (js/json)

File: src/components/FormOfSeach/FormFieldSearchHotel/FormFieldSearchHotel.js [hunk 11]
```diff
@@ -1,46 +1,67 @@
...
+                name
+            </StyledLabelFist>
+            <StyledInputFist
```
Comment (ghost) [inline src/components/FormOfSeach/FormFieldSearchHotel/FormFieldSearchHotel.js]: fix typos, please

File: src/components/FormOfSeach/FormFieldSearchHotel/FormFieldSearchHotel.js [hunk 12]
```diff
@@ -1,46 +1,67 @@
...
+                onFocus={handleFocus}
+                onBlur={disableFocus}
+                name='search'
                 type='text'
                 id='searchName'
-                className='form__input'
-                value={values.name}
-                onChange={handleChange}
+                value={values}
+                onChange={(e) => {
```
Comment (ghost) [inline src/components/FormOfSeach/FormFieldSearchHotel/FormFieldSearchHotel.js:56]: move this function into separate handler

File: src/components/FormOfSeach/FormFieldWhoAreGoing/FormFieldWhoAreGoing.js [hunk 13]
```diff
@@ -1,64 +1,75 @@
...
-import "./FormFieldWhoAreGoing.css"
 import Counter from "./countsOrder/counterOrders"
 import OrderData from "./dataOrder/dataOrder"
 import Select from "./childrenSelects/childrenSelect";
+import uniqid from 'uniqid';
+import {useSelector} from "react-redux";
+import {ClickAwayListener} from "@mui/material";
+import {ModalForm, SelectWrapper, DataFormWrapper, ModalFormQuestion} from "../../Styled-components/HeaderForm";
 
 const ModalMenu = () => {
```
Comment (ghost) [inline src/components/FormOfSeach/FormFieldWhoAreGoing/FormFieldWhoAreGoing.js]: looks like a custom hook. apply hooks-name-styling, starting from `use` (useDispatch, useSelector etc.)

File: src/components/FormOfSeach/FormFieldWhoAreGoing/FormFieldWhoAreGoing.js [hunk 14]
```diff
@@ -1,64 +1,75 @@
...
         }
     )
 }
 
 const FormFieldThird = () => {
-    const {data, countOfSelect} = OrderData()
-    const {isOpen, handleClick} = ModalMenu()
+    const selectArray = useSelector(state => state.formReducer.children)
+    const {data} = OrderData()
+    const {isOpen, handleClick, handleClickAway} = ModalMenu()
```
Comment (ghost) [inline src/components/FormOfSeach/FormFieldWhoAreGoing/FormFieldWhoAreGoing.js]: the same

File: src/components/FormOfSeach/FormFieldWhoAreGoing/countsOrder/counterOrders.js [hunk 15]
```diff
@@ -1,27 +1,41 @@
...
         <>
-            <div className="counterWrapper">
-                <div className='counterLabel'>{`${label}`}</div>
-                <div className='counterChanger'>
-                    <button className={
-                        count === minValue ? 'btn' : 'active btn'
-                    } onClick={onDecrement}>-
-                    </button>
+            <ModalFormCounterWrapper>
+                <div>{`${label}`}</div>
```
Comment (ghost) [inline src/components/FormOfSeach/FormFieldWhoAreGoing/countsOrder/counterOrders.js]: `<div>{label}</div>`

File: src/components/FormSingUp/Form.jsx [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+        event.preventDefault();
+    }
+    return (
+        <FormLogin onSubmit={handleSubmit}>
+            <label>
+                Email address
+                <input
+                    type="email"
+                    value={email}
+                    onChange={(e) => setEmail(e.target.value)}
```
Comment (ghost) [inline src/components/FormSingUp/Form.jsx]: move it into separate handler

File: src/components/FormSingUp/Form.jsx [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+                />
+            </label>
+            <label>
+                Password
+                <input
+                    type="password"
+                    value={pass}
+                    onChange={(e) => setPass(e.target.value)}
```
Comment (ghost) [inline src/components/FormSingUp/Form.jsx]: the same

File: src/components/FormSingUp/Form.jsx [hunk 18]
```diff
@@ -0,0 +1,37 @@
...
+                />
+            </label>
+            <button
+                onClick={() => handleClick(email, pass)}
```
Comment (ghost) [inline src/components/FormSingUp/Form.jsx]: the same

File: src/components/HomesGuestsLovesSlider/HomesGuestsLovesSlider.js [hunk 19]
```diff
@@ -1,23 +1,90 @@
...
+                    </SliderArrow>
+                </SliderButtonPrev>
             </div>
-        );
+        </div>
+    );
+}
+
+const HomesGuestsLovesSlider = () => {
+    const {array} = useFeatch();
```
Comment (ghost) [inline src/components/HomesGuestsLovesSlider/HomesGuestsLovesSlider.js]: fix typo

File: src/components/HotelPageFetch/HotelPageFetch.js [hunk 20]
```diff
@@ -1,10 +1,14 @@
 import {useParams} from "react-router-dom";
-import {useEffect, useState} from "react";
+import {useEffect, useState,} from "react";
 
 const HotelPageFetch = () => {
     const {id} = useParams()
     const url = `https://fe-student-api.herokuapp.com/api/hotels/${id}`;
```
Comment (ghost) [inline src/components/HotelPageFetch/HotelPageFetch.js]: why do you store static path here in the component?

File: src/hooks/useFetch.js [hunk 21]
```diff
@@ -1,6 +1,7 @@
 import {useEffect, useState} from "react";
 
-function useFeatch(url) {
+const useFeatch = () => {
+    const url = "https://fe-student-api.herokuapp.com/api/hotels/popular";
```
Comment (ghost) [inline src/hooks/useFetch.js]: the same - it would be better to pass url throw params

File: src/redux/actions.js [hunk 22]
```diff
@@ -0,0 +1,107 @@
...
+export const loadingData = (loading) => {
+    return {
+        type: LOADING_DATA,
+        loading
+    };
+};
+
+export const hotelsLoad = (params) => {
+    return async dispatch => {
+        const response = await fetch(`https://fe-student-api.herokuapp.com/api/hotels?${params}`);
```
Comment (ghost) [inline src/redux/actions.js]: the same - static urls

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/components/AvailableHotelsSlider/AvailableHotelsSlider.js (hunk 5 above)
Comment (ArtyomLobanok) [reply src/components/AvailableHotelsSlider/AvailableHotelsSlider.js]: don't need, i deleted it

File: src/components/AvailableHotelsSlider/AvailableHotelsSlider.js (hunk 4 above)
Comment (ArtyomLobanok) [reply src/components/AvailableHotelsSlider/AvailableHotelsSlider.js]: did this

File: src/components/AvailableHotelsSlider/AvailableHotelsSlider.js (hunk 6 above)
Comment (ArtyomLobanok) [reply src/components/AvailableHotelsSlider/AvailableHotelsSlider.js]: did this

File: src/components/FormOfSeach/Form.js (hunk 8 above)
Comment (ArtyomLobanok) [reply src/components/FormOfSeach/Form.js]: did it

File: src/components/FormOfSeach/FormFieldSearchHotel/FormFieldSearchHotel.js (hunk 12 above)
Comment (ArtyomLobanok) [reply src/components/FormOfSeach/FormFieldSearchHotel/FormFieldSearchHotel.js:56]: did this

File: src/components/FormOfSeach/FormFieldSearchHotel/FormFieldSearchHotel.js (hunk 11 above)
Comment (ArtyomLobanok) [reply src/components/FormOfSeach/FormFieldSearchHotel/FormFieldSearchHotel.js]: did this

File: src/components/FormOfSeach/FormFieldWhoAreGoing/FormFieldWhoAreGoing.js (hunk 13 above)
Comment (ArtyomLobanok) [reply src/components/FormOfSeach/FormFieldWhoAreGoing/FormFieldWhoAreGoing.js]: did this

File: src/components/FormOfSeach/FormFieldWhoAreGoing/FormFieldWhoAreGoing.js (hunk 14 above)
Comment (ArtyomLobanok) [reply src/components/FormOfSeach/FormFieldWhoAreGoing/FormFieldWhoAreGoing.js]: did this

File: src/components/FormOfSeach/FormFieldWhoAreGoing/countsOrder/counterOrders.js (hunk 15 above)
Comment (ArtyomLobanok) [reply src/components/FormOfSeach/FormFieldWhoAreGoing/countsOrder/counterOrders.js]: did this

File: src/components/HomesGuestsLovesSlider/HomesGuestsLovesSlider.js (hunk 19 above)
Comment (ArtyomLobanok) [reply src/components/HomesGuestsLovesSlider/HomesGuestsLovesSlider.js]: did it

File: src/components/FormSingUp/Form.jsx (hunk 16 above)
Comment (ArtyomLobanok) [reply src/components/FormSingUp/Form.jsx]: did this

File: src/components/FormSingUp/Form.jsx (hunk 17 above)
Comment (ArtyomLobanok) [reply src/components/FormSingUp/Form.jsx]: did this

File: src/components/HotelPageFetch/HotelPageFetch.js (hunk 20 above)
Comment (ArtyomLobanok) [reply src/components/HotelPageFetch/HotelPageFetch.js]: did this

File: src/hooks/useFetch.js (hunk 21 above)
Comment (ArtyomLobanok) [reply src/hooks/useFetch.js]: did this

File: src/redux/actions.js (hunk 22 above)
Comment (ArtyomLobanok) [reply src/redux/actions.js]: did this

File: src/components/FormOfSeach/FormFieldSearchHotel/FormFieldSearchHotel.js (hunk 10 above)
Comment (ArtyomLobanok) [reply src/components/FormOfSeach/FormFieldSearchHotel/FormFieldSearchHotel.js]: did this

**Pull request merged**